### PR TITLE
Fix popupmenu position in multigrid mode (pum_above)

### PR DESCRIPTION
**Pull request by last-partizan**
This fixes issue https://github.com/neovim/neovim/issues/12985

### CODE REVIEW

File: test/functional/ui/float_spec.lua [hunk 1]
```diff
@@ -2901,6 +2901,56 @@ describe('float window', function()
...
+            {12:~           }|
+          ## grid 5
+            {13: aa             }|
+            {1: word           }|
+            {1: longtext       }|
+          ]], float_pos={
+            [4] = {{id = 1001}, "NW", 1, 2, 5, true, 50 },
+            -- next line is still incorrect, i expect SW to be there,
+            -- but actual result is NW, and grid doesn't look right
+            [5] = {{id = -1}, "SW", 4, 3, 1, false, 100 },
```
Comment (last-partizan) [inline test/functional/ui/float_spec.lua:2939]: @bfredl please take a look at this.

`screen:expect` without multigrid looking fine, but with multigrid i exepct here to be SW, but actual result is NW.

any ideas what i doing wrong here?

Comment (bfredl) [reply test/functional/ui/float_spec.lua:2939]: Is it wrong tho? It just seems it decided to place the popupmenu below. I'd suggest using the cmdline as in your original screenshot, to add a test which takes the SW branch.

Comment (last-partizan) [reply test/functional/ui/float_spec.lua]: That's nice idea.

Please take a look, i updated the test and it does what i expect.

**Comment by bfredl** [comment]
looks good. force pushed to restart GHA.

**Pull request merged**